src/proteus/dswx_hls.py

```diff
@@ -2355,6 +2498,74 @@ def _save_output_rgb_file(red, green, blue, output_file,
     logger.info(f'file saved: {output_file}')
 
 
+def _compute_browse_array(
+        masked_interpreted_water_layer,
+        flag_collapse_wtr_classes=FLAG_COLLAPSE_WTR_CLASSES,
+        include_psw_aggressive=False,
+        set_not_water_to_nodata=False,
+        set_cloud_to_nodata=False,
+        set_snow_to_nodata=False):
+    """
+    Generate a version of the WTR-2 layer where the
+    pixels marked with cloud and snow in the CLOUD layer
+    are designated with unique values per dswx_hls.py constants (see notes).
+
+    Parameters
+    ----------
+    masked_interpreted_water_layer : numpy.ndarray
+        CLOUD-masked interpreted water layer
+        (i.e. the DSWx-HLS WTR layer)
+    flag_collapse_wtr_classes : bool
+        Collapse interpreted layer water classes following standard
+        DSWx-HLS product water classes
+    include_psw_aggressive : bool
+        True to include Partial Surface Water Aggressive class (PSW-Agg)
+        in output layer. False to not display these pixels as PSW and instead
+        display them as Not Water. Default is False.
+        If `flag_collapse_wtr_classes` is False, then `include_psw_aggressive`
+        will be ignored and PWS-Agg will be displayed.
+    set_not_water_to_nodata : bool
+        How to code the Not Water pixels. Defaults to False. Options are:
+            True : Not Water pixels will be marked with UINT8_FILL_VALUE
+            False : Not Water will remain WTR_NOT_WATER
+    set_cloud_to_nodata : bool
+        How to code the cloud pixels. Defaults to False. Options are:
+            True : cloud pixels will be marked with UINT8_FILL_VALUE
+            False : cloud will remain WTR_CLOUD_MASKED
+    set_snow_to_nodata : bool
+        How to code the snow pixels. Defaults to False. Options are:
+            True : snow pixels will be marked with UINT8_FILL_VALUE
+            False : snow will remain WTR_CLOUD_MASKED_SNOW
+    
+    Returns
+    -------
+    browse_arr : numpy.ndarray
+        Interpreted water layer adjusted for the input parameters provided.
+    """
+
+    # Create a copy of the masked_interpreted_water_layer.
+    browse_arr = masked_interpreted_water_layer.copy()
+
+    # Discard the Partial Surface Water Aggressive class
+    if not include_psw_aggressive and flag_collapse_wtr_classes:
+        browse_arr[browse_arr == WTR_UNCOLLAPSED_LOW_CONF_WATER] = \
+                                                            WTR_NOT_WATER
+
+    if flag_collapse_wtr_classes:
+        browse_arr = _collapse_wtr_classes(browse_arr)
```

Comment:
The code actually uses `_save_array()` to save the browse image; it does not use `save_dswx_product()`. The browse image uses `_save_arr()` because it needs a custom color table.

Reply:
Oh, I see. Good point. I forgot to mention that you may want to add a description to the browse .tif file. But we can do that later too (in another PR).

Reply:
Sounds good.

And if we remember to add the description, that's fine. But the BROWSE.tif is only being used by DAAC for creating add'l images, and it will not be included in the final granules for the users, so it's probably not necessary?

Reply:
Yes, it's not necessary, but a small thing that would be nice to add at some point, doesn't need to be now.